codes/classical/q-ary_digits/gabidulin.yml

```diff
@@ -0,0 +1,29 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: gabidulin
+physical: q-ary_digits
+logical: q-ary_digits
+
+name: 'Gabidulin or rank code'
+introduced: '\cite{manual_citation_minilatex:{E. M. Gabidulin, "Theory of Codes with Maximum Rank Distance", Problemy Peredachi Informacii, Volume 21, Issue 1, \emph{3–16} (1985)}}'
+
+description: 'Error-correcting codes over rank metric.'
+
+protection: '\(d\)-distance codes protect agains all errors of rank not greater \(\lfloor (d-1)/2\rfloor\)'
+
+notes:
+ - |
+   \emph{Rank metric.} For \(X^n\), n-dimensional vector space over \(GF(q^N)\) where \(q\) is a power of a prime number,
+   rank metric \(d(x, y)\) is defined via rank norm 
+   \(r(x, q):= \mathrm{rank}(A(x))\) where 
+   \(A(x) = \begin{pmatrix} a_{11} & a_{12} & \ldots & a_{1n} \\ a_{21} & a_{22} & \ldots & a_{2n} \\  a_{N1} & a_{N2} & \ldots & a_{Nn} \end{pmatrix} \)
+   and \(x_i = a_{1i} u_1 + a_{2i} u_2 + \ldots + a_{Ni}u_N \) for some fixed basis \(\{u_i\}_{i=1}^N\).
+ - 'Set of vectors \(\{x_1, x_2, \ldots, x_M\}\) is called a \emph{code} with distance \(d=\min d(x_i, x_j)\). If the set defines a \(k\)-dimensional subspace, then the code is [n,k]-linear.'
+
```

Comment:
realizations:
    'Wikipedia says something about network coding?

Reply:
Not sure what details are needed here